storage/src/robocorp/storage/__init__.py

```diff
@@ -22,142 +22,128 @@ def list_assets() -> List[AssetMeta]:
-        nonlocal exception
         try:
             assets_client.handle_error(resp)
         except RequestsHTTPError as exc:
             if exc.status_code == 404:
-                exception = exc
+                message = f"Asset {name!r} with id {asset_id!r} not found"
+                raise AssetNotFound(message) from exc
             else:
-                raise
+                raise exc
 
-    LOGGER.debug("Retrieving asset %r with resulted ID %r.", name, asset_id)
     response = assets_client.get(asset_id, _handle_error=_handle_error)
-    if response.ok:
-        return cast(Asset, response.json())
-
-    message = f"asset with name {name!r} and resulted ID {asset_id!r} couldn't be found"
-    if raise_if_missing:
-        raise AssetNotFound(message) from exception
-
-    LOGGER.warning("%s%s!", message[0].upper(), message[1:])
-    return None
+    return cast(Asset, response.json())
 
 
 def get_asset(name: str) -> str:
-    """Get the asset's value by providing its `name`.
+    """Get an asset's value by providing its `name`.
 
     Args:
-        name: Name of the asset to fetch.
+        name: Name of the asset
 
     Returns:
-        The previously set value of this asset. (empty if it was never set)
+        The previously set value of this asset, or empty string if not set
 
     Raises:
-        AssetNotFound: When the queried asset doesn't exist.
+        AssetNotFound: Asset with given name does not exist
     """
-    LOGGER.info("Retrieving asset %r.", name)
-    payload = _get_asset(name)["payload"]  # type: ignore
+    LOGGER.info("Retrieving asset: %r", name)
+    payload = _get_asset(name)["payload"]
     if payload["type"] == "empty":
-        LOGGER.warning("Asset %r has no value set!", name)
+        LOGGER.warning("Asset %r has no value set", name)
         return ""
 
     assets_client = _get_assets_client()
     url = payload["url"]
     return assets_client.get(url, headers={}).text
 
 
-def _create_asset(name: str):
-    LOGGER.debug("Creating new asset with name %r.", name)
+def _create_asset(name: str) -> Asset:
+    LOGGER.debug("Creating new asset with name: %r", name)
     assets_client = _get_assets_client()
     body = {"name": name}
     return assets_client.post("", json=body).json()
 
 
 def set_asset(name: str, value: str, wait: bool = True):
-    """Sets/Creates an asset named `name` with the provided `value`.
+    """Creates or updates an asset named `name` with the provided `value`.
 
     Args:
-        name: Name of the existing or new asset to create. (if missing)
-        value: The new value set within the asset.
-        wait: Blocks until the new value is reflected within the asset.
+        name: Name of the existing or new asset to create (if missing)
+        value: The new value set within the asset
+        wait: Wait for value to be set succesfully
 
     Raises:
-        AssetNotFound: When the queried asset doesn't exist.
+        RuntimeError: Asset upload failed
```

Comment:
Good catch! What do you think here of a custom exception, instead of relying on Python's ones? We can check with @MarioQuiroga32 in what cases (like content type reason) an API upload would fail, so we know best from what to derive, `RuntimeError` vs. `ValueError`.